roles/master/tasks/main.yml

```diff
@@ -1,3 +1,74 @@
+- name: add Mercurial yum repo
+  yum_repository:
+    name: mercurial.selenic.com
+    description: Mercurial
+    baseurl: https://www.mercurial-scm.org/release/centos7/
+    gpgcheck: no
+
+- name: install dev programs to compile cloud-provider-openstack
+  yum:
+    disable_gpg_check: yes
+    state: present
+    name:
+      - git
+      - python-devel
+      - gcc
+      - gcc-c++
+      - kernel-devel
+      - mercurial
+      - python-pip
+
+- name: ensure go module source directories exist
+  become: no
+  file:
+    path: /home/centos/go/src
+    state: directory
+
+- name: check if docker cephfs image exists
+  command: >
```

Comment:
This needs to be shell module rather than command module b/c it is a pipeline.

Debug shows it running a single, successful grep command.

Or we may be able to use command module with:
     docker image inspect quay.io/cephcsi/cephfsplugin:v1.0.0

Reply:
Oh... I thought the grep here would fail..

Reply:
There are a bunch of greps and debug shows '0' return code.  Maybe it's getting that from the first 'grep -v' command.  I'm testing with the docker inspect command (which web search indicates shoule return non-zero if it can't match and zero if it does match) now.

Reply:
This worked:

- name: check if docker cephfs image exists
  command: >
     docker image inspect quay.io/cephcsi/cephfsplugin:v1.0.0
  register: result
  ignore_errors: True

- name: make the cephfs image and push it to the local registry
  become: no
  shell: |
    source /etc/profile.d/golang.sh
    go get -d github.com/ceph/ceph-csi
    cd /home/centos/go/src/github.com/ceph/ceph-csi
    git checkout master
    make image-cephfsplugin
  when: result is failed

Reply:
My formatting was screwed up.  https://paste.fedoraproject.org/paste/-FmNUhe4I4~AucwZAv1qcA

Reply:
We can 'become: no' here, right?

Reply:
grr, commented elsewhere. Ack on both issues. Real response here: https://github.com/tombarron/manila-kube/pull/11#pullrequestreview-230297167